### ISSUE TITLE
missing a bunch of the "classification" functions

### ISSUE THREAD

**Issue by silky** [tensorflow/haskell#10]
from here: https://www.tensorflow.org/versions/r0.11/api_docs/python/nn.html#classification

i.e. i'd like `sigmoid_cross_entropy_with_logits`, it's not there

glancing around it appears that there are many ops that are in `tensorflow/python/ops/nn.py` that aren't defined here; i guess because they can't be straightfowardly gen'd like the ones in `nn_ops.py`.

do the maintainers have some sort of plan for these? or should I just go about adding them as PRs as i happen on ones I care about?

**Comment by blackgnezdo** [comment]
@silky, you are in luck, I happen to have written this one :) Would you be willing to start a new library, maybe tensorflow-nn, and put this code there? I'll be happy to review a PR (you'll have to sign the CLA, it's generally quick):

sigmoidCrossEntropyWithLogits
  :: (TF.OneOf '[Float, Double] a, TF.TensorType a, Num a) =>
     TF.Tensor TF.Value a
     -> TF.Tensor TF.Value a -> TF.Build (TF.Tensor TF.Value a)
sigmoidCrossEntropyWithLogits logits targets = do
    let zeros = TF.zerosLike logits
        cond = logits `TF.greaterEqual` zeros
        relu_logits = TF.select cond logits zeros
        neg_abs_logits = TF.select cond (-logits) logits
    TF.withNameScope "logistic_loss" $ do
        left <- TF.render $ relu_logits - logits * targets
        right <- TF.render $ TF.log (1 + TF.exp neg_abs_logits)
        TF.withNameScope "sigmoid_add" $ TF.render $ left `TF.add` right
